### PR TITLE
feat(arbitrary): add suport for `f64`

### LINKED ISSUE #1208: [Feature] Add support for `f64` in `alloy-serde` arbitrary module
### Component

serde

### Describe the feature you would like

Add de-/serialization of `f64` from/to `String` in module _crates/serde/src/other/arbitrary\_.rs_

https://github.com/alloy-rs/op-alloy/blob/f9514d5914abc6d2f4979db59d6081a60e8adfc7/crates/rpc-types/src/receipt.rs#L64-L86

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mvares**
Closes #1208

### CODE REVIEW

**Comment by mvares** [comment]
can you check out? 

cc @emhane 

PS: I added the `from_json_value` method due to deserialization.

File: crates/serde/src/other/arbitrary_.rs [hunk 1]
```diff
@@ -31,6 +32,7 @@ impl ArbitraryValue {
         match self {
             Self::Null => serde_json::Value::Null,
             Self::Bool(b) => serde_json::Value::Bool(b),
+            Self::Float(n) => serde_json::Value::String(n.to_string()),
```
Comment (mattsse) [inline crates/serde/src/other/arbitrary_.rs:35]: we should be able to convert this into Value::Number via `Number::from_f64`

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/serde/src/other/arbitrary_.rs (hunk 1 above)
Comment (mvares) [reply crates/serde/src/other/arbitrary_.rs:35]: WDYT?
`Self::Float(n) => serde_json::Number::from_f64(n).into()`

Comment (mattsse) [reply crates/serde/src/other/arbitrary_.rs:35]: that should work, we can treat the None case as 0.0 that should be okay given this is only used for testing

**Review by emhane — COMMENTED** [review]
taking @mattsse 's review as a confirmation of the issue. i added the 'needs triage' label to the issue since I wasn't sure if ethereum stopped using f64 and hence if it made sense to impl this. OP uses `f64` for the `l1FeeScalar`, but that value is depreciated since ecotone.

**Comment by mattsse** [comment]
I see, then I think this isn't that useful

**Pull request closed without merging**